client/backend/src/lib.rs

```diff
@@ -1,218 +1,78 @@
-            ServerMessage::Response {
-                response,
-                request_id: response_id,
-            } => {
-                match response {
-                    // TODO do this more asynchronously @gegy1000
-                    RequestResponse::Success(Success::Room { id }) if response_id == request_id => {
-                        Ok(id)
-                    }
-                    RequestResponse::Error(e) => Err(Error::ServerError(e)),
-                    _ => Err(Error::IncorrectServerMessage(msg)),
-                }
-            }
-            ServerMessage::Error(e) => Err(Error::ServerError(e)),
-            _ => Err(Error::IncorrectServerMessage(msg)),
-        }
+    pub fn create_room(&mut self) {
+        self.net.send(ServerboundMessage::CreateRoom);
     }
 
-    pub fn join_room(&mut self, room: RoomId) -> Result<(), Error> {
-        let request_id = self.request(ClientMessage::JoinRoom(room))?;
-
-        let msg = self.receive_blocking()?;
-        match msg.clone() {
-            ServerMessage::Response {
-                response,
-                request_id: response_id,
-            } => {
-                match response {
-                    // TODO do this more asynchronously @gegy1000
-                    RequestResponse::Success(Success::NoData) if response_id == request_id => {
-                        Ok(())
-                    }
-                    RequestResponse::Error(e) => Err(Error::ServerError(e)),
-                    _ => Err(Error::IncorrectServerMessage(msg)),
-                }
-            }
-            ServerMessage::Error(e) => Err(Error::ServerError(e)),
-            msg @ _ => Err(Error::IncorrectServerMessage(msg)),
-        }
+    pub fn join_room(&mut self, room: RoomId) {
+        self.net.send(ServerboundMessage::JoinRoom(room));
     }
 
     /// Sends a message, returning the request id if it was sent successfully
-    pub fn send_message(&mut self, msg: String, to_room: RoomId) -> Result<RequestId, Error> {
-        if !self.logged_in {
-            self.request(ClientMessage::SendMessage(ClientSentMessage {
-                to_room,
-                content: msg,
-            }))
-        } else {
-            Err(Error::NotLoggedIn)
-        }
+    pub fn send_message(&mut self, content: String, to_room: RoomId) {
+        self.net.send(ServerboundMessage::SendMessage(ClientSentMessage { to_room, content }));
     }
 
-    /// Should be called once every `HEARTBEAT_INTERVAL`
-    pub fn heartbeat(&mut self) -> Result<(), Error> {
-        self.socket
-            .send_message(&OwnedMessage::Ping(vec![]))
-            .map_err(Error::WebSocketError)
+    pub fn username(&self) -> String {
+        format!("{}", self.user_id.0) // TODO lol
     }
 
-    pub fn username(&self) -> String {
-        format!("{}", self.id.0) // TODO lol
+    /// Should be called once every `HEARTBEAT_INTERVAL`
+    #[inline]
+    pub fn dispatch_heartbeat(&mut self) {
```

Comment:
`dispatch`?

Reply:
dispatch is to 'send off to a destination or for a purpose'

Reply:
Just thought it was strange phrasing, but alright